### PR TITLE
Ensure link names are unique for accessibility, thanks @nickvergessen, fix #575

### LINKED ISSUE #575: Duplicate links on page with generic title "Open file"
Currently there are a lot of duplicate links on the page all saying "Open file", which makes it a problem for screen readers:
image

We have 2 ways of fixing this:
- We could change the text to "Open `filename`" stating the actual filename. But actually the links are already there in text form directly above. So I would rather propose:
- We change the previews into div elements rather than links so they are not picked up by screenreaders and not read out duplicate. We can still attach JS event to them so on click the file gets opened, but then if you keyboard-tab through the page you won’t get hit by duplicates.

### PR DESCRIPTION
**Pull request by jancborchardt**
Before, there were a lot of duplicate links on the page all saying "Open file", which makes it a problem for screen readers:
image

This is now fixed by changing the link to say "Open `filename`" stating the actual filename.

Since the links are already there in text form directly above, in the future we could also remove the duplicate linked images from screenreader flow by using divs for them and triggering the link via Javascript.

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -253,7 +253,7 @@ $(function(){
 				for (var i = 0; i < activity.previews.length; i++) {
 					var preview = activity.previews[i];
 					content += ((preview.link) ? '<a href="' + preview.link + '">' + "\n" : '')
-						+ '<img class="preview' + ((preview.isMimeTypeIcon) ? ' preview-mimetype-icon' : '') + '" src="' + preview.source + '" alt="' + t('activity', 'Open file') + '" />' + "\n"
+						+ '<img class="preview' + ((preview.isMimeTypeIcon) ? ' preview-mimetype-icon' : '') + '" src="' + preview.source + '" alt="' + t('activity', 'Open ') + preview.filename + '" />' + "\n"
```
Comment (nickvergessen) [inline js/script.js:256]:
```suggestion
						+ '<img class="preview' + ((preview.isMimeTypeIcon) ? ' preview-mimetype-icon' : '') + '" src="' + preview.source + '" alt="' + t('activity', 'Open {filename}', preview) + preview.filename + '" />' + "\n"
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
Pushed a fix to the unit tests

File: lib/Controller/APIv2Controller.php [hunk 2]
```diff
@@ -363,6 +363,7 @@ protected function getPreview(string $owner, int $fileId, string $filePath): arr
 			'isMimeTypeIcon' => true,
 			'fileId' => $fileId,
 			'view' => $info['view'] ?: 'files',
+			'filename' => basename($filePath),
```
Comment (MorrisJobke) [inline lib/Controller/APIv2Controller.php:366]: @nickvergessen This is an REST API change, shouldn't it be documented?

Comment (nickvergessen) [reply lib/Controller/APIv2Controller.php:366]: @jancborchardt can you add it in https://github.com/nextcloud/activity/blob/master/docs/endpoint-v2.md#preview-element ? :P

Comment (MorrisJobke) [reply lib/Controller/APIv2Controller.php:366]: https://github.com/nextcloud/activity/pull/588

**Pull request merged**